test/bdd/mock/loginconsent/main.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/trustbloc/edge-core/pkg/log"
+	"github.com/trustbloc/edge-core/pkg/storage"
+	"github.com/trustbloc/edge-core/pkg/storage/memstore"
+	tlsutils "github.com/trustbloc/edge-core/pkg/utils/tls"
+)
+
+var logger = log.New("mock-login-consent")
+
+type config struct {
+	listenAddr    string
+	hydraAdminURL *url.URL
+	tlsConfig     *tls.Config
+	serveCertFile string
+	serveKeyFile  string
+	store         storage.Store
+}
+
+func main() {
+	conf, err := loadConfig()
+	if err != nil {
+		logger.Errorf("failed to load config: %s", err.Error())
+		os.Exit(1)
+	}
+
+	logger.Infof("initializing server on %s", conf.listenAddr)
+
+	err = http.ListenAndServeTLS(
+		conf.listenAddr,
+		conf.serveCertFile,
+		conf.serveKeyFile,
+		newServer(conf),
+	)
+	if err != nil {
+		logger.Errorf("server encountered an error: %s", err.Error())
+		os.Exit(1)
+	}
+}
+
+func loadConfig() (*config, error) {
+	hydraAdminURL, err := url.Parse(os.Getenv("HYDRA_ADMIN_URL"))
+	if err != nil {
+		return nil, fmt.Errorf("env variable HYDRA_ADMIN_URL missing or malformed: %w", err)
+	}
+
+	serveCertPath := os.Getenv("TLS_CERT_PATH")
+	if serveCertPath == "" {
+		return nil, fmt.Errorf("env variable TLS_CERT_PATH required")
+	}
+
+	serveKeyPath := os.Getenv("TLS_KEY_PATH")
+	if serveKeyPath == "" {
+		return nil, fmt.Errorf("env variable TLS_KEY_PATH required")
+	}
+
+	rootCaCertsPath := os.Getenv("ROOT_CA_CERTS_PATH")
+	if rootCaCertsPath == "" {
+		return nil, fmt.Errorf("env variable ROOT_CA_CERTS_PATH required")
+	}
+
+	rootCACerts, err := tlsutils.GetCertPool(false, []string{rootCaCertsPath})
+	if err != nil {
+		return nil, fmt.Errorf("failed to init tls cert pool from path %s: %w", rootCaCertsPath, err)
+	}
+
+	listenAddr := os.Getenv("LISTEN_ADDR")
+	if listenAddr == "" {
+		return nil, fmt.Errorf("env variable LISTEN_ADDR required")
+	}
+
+	prov := memstore.NewProvider()
+
+	err = prov.CreateStore("bdd_tests")
+	if err != nil {
+		return nil, fmt.Errorf("could not create memory store for some reason: %w", err)
```

Comment:
The "for some reason" part sounds sort of odd/casual for an error message to print out. You can just say "could not create memory store: %w"

Reply:
I used this code from hub-auth

Reply:
But I agree the msg is wrong

Reply:
done